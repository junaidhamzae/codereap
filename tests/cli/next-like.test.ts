import path from 'node:path';
import fs from 'node:fs';
import os from 'node:os';
import { runCli } from '../helpers/cli';

function getRow(json: any[], endsWith: string) {
  return (json as any[]).find((r) => r.node.endsWith(endsWith));
}

describe('CLI Next-like auto seeding', () => {
  it('prints Next detection and seeds entrypoints', () => {
    const fixture = path.resolve(__dirname, '../fixtures/next-like');
    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-'));
    const outBase = path.join(tmp, 'report');
    const res = runCli(['--root', fixture, '--out', outBase]);
    expect(res.status).toBe(0);
    // page should be live, and component reachable
    const rows = res.json as any[];
    expect(getRow(rows, 'pages/index.ts').orphan).toBe(false);
    expect(getRow(rows, 'components/Widget.ts').orphan).toBe(false);
    // stdout should include Next detection message
    expect(res.stdout).toMatch(/Next\.js detected/);
    expect(rows).toMatchSnapshot();
  });
});


