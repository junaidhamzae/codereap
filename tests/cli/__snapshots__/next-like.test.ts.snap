// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CLI Next-like auto seeding prints Next detection and seeds entrypoints 1`] = `
[
  {
    "exists": true,
    "in-degree": 1,
    "node": "components/Widget.ts",
    "orphan": false,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "Widget",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "package.json",
    "orphan": true,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
      "imports": [],
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "pages/index.ts",
    "orphan": false,
    "symbols": {
      "exports": {
        "default": {
          "exists": true,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
    },
  },
]
`;
